src/lib.rs

```diff
@@ -1090,4 +1090,18 @@ mod tests {
             &["this.associated", "foo.associated", "this-associated"],
         );
     }
+
+    #[test]
+    fn invalid_duplicate_inherit() {
+        error_policy_test!("duplicate_inherit.cas", 2, ErrorItem::Compile(_));
+    }
+
+    #[test]
+    fn valid_duplicate_inherit() {
+        valid_policy_test("duplicate_inherit.cas", &[
```

Comment:
I built that cas file with what is on main and "typeattributeset foo (qux)" is missing.  I agree this might be a short coming with the compiler.  But I haven't mucked around with the cil stuff enough to say if having "(typeattributeset foo (baz))" & "(typeattributeset baz (qux))" is good enough.

Reply:
Okay, I read the docs and poked at secilc a bit, and you are absolutely right.  CIL handles this for us via transitive typeattribueset.  So this is a non-issue.